src/test/rpc/KeyGeneration_test.cpp

```diff
@@ -709,6 +707,72 @@ class WalletPropose_test : public ripple::TestSuite
         }
     }
 
+    void testRippleLibEd25519()
+    {
+        testcase ("ripple-lib encoded Ed25519 keys");
+
+        auto test = [this](char const* seed, char const* addr)
+        {
+            {
+                Json::Value params;
+                Json::Value error;
+
+                params[jss::passphrase] = seed;
+
+                auto ret = keypairForSignature(params, error);
+
+                BEAST_EXPECT(!contains_error(error));
+                BEAST_EXPECT(ret.first.size() != 0);
+                BEAST_EXPECT(toBase58(calcAccountID(ret.first)) == addr);
+            }
+            
+            {
+                Json::Value params;
+                Json::Value error;
+
+                params[jss::key_type] = "ed25519";
+                params[jss::seed] = seed;
+
+                auto ret = keypairForSignature(params, error);
+
+                BEAST_EXPECT(!contains_error(error));
+                BEAST_EXPECT(ret.first.size() != 0);
+                BEAST_EXPECT(toBase58(calcAccountID(ret.first)) == addr);
+            }
+
+            {
+                Json::Value params;
+                Json::Value error;
+
+                params[jss::key_type] = "secp256k1";
+                params[jss::secret] = seed;
```

Comment:
This should be `jss::seed`, not `jss::secret`. A `jss::secret` is not allowed to set the key type, and this test is testing for a key type mismatch.

Reply:
:+1: